### PR TITLE
Do not erroneously map the escape character onto the up key

### PR DESCRIPTION
**Pull request by exg**

### CODE REVIEW

File: yank.c [hunk 1]
```diff
@@ -316,18 +316,19 @@ tgetc(void)
...
 		{ NULL,		0 },
 	};
-	char	buf[3];
+	char	buf[4];
 	ssize_t	n;
 	int	i;
 
-	n = read(tty.rfd, buf, sizeof(buf));
+	n = read(tty.rfd, buf, sizeof(buf) - 1);
+	buf[n] = 0;
```
Comment (mptre) [inline yank.c]: Caution here, `n` can be negative. Put this after `return KEY_TERM`. Also favor `'\0'` over `0`.

**Review by mptre — CHANGES_REQUESTED** [review]
Nice find. Some changes must be done before this goes in.

File: yank.c (hunk 1 above)
Comment (exg) [reply yank.c]: Ah sorry I overlooked that. Fixed.

**Comment by mptre** [comment]
Merged, thanks!

**Pull request merged**